src/wasapi.c

```diff
@@ -863,17 +917,26 @@ static int refresh_devices(struct SoundIoPrivate *si) {
         if (FAILED(hr = IPropertyStore_GetValue(rd.prop_store,
                         PKEY_DEVICE_FRIENDLYNAME, &rd.prop_variant_value)))
         {
-            deinit_refresh_devices(&rd);
-            return SoundIoErrorOpeningDevice;
+            rd.device_shared->probe_error = SoundIoErrorOpeningDevice;
+            rd.device_raw->probe_error = SoundIoErrorOpeningDevice;
+            rd.device_shared = NULL;
+            rd.device_raw = NULL;
+            continue;
         }
         if (!rd.prop_variant_value.pwszVal) {
-            deinit_refresh_devices(&rd);
-            return SoundIoErrorOpeningDevice;
+            rd.device_shared->probe_error = SoundIoErrorOpeningDevice;
+            rd.device_raw->probe_error = SoundIoErrorOpeningDevice;
+            rd.device_shared = NULL;
+            rd.device_raw = NULL;
+            continue;
         }
         int device_name_len;
         if ((err = from_lpwstr(rd.prop_variant_value.pwszVal, &rd.device_shared->name, &device_name_len))) {
-            deinit_refresh_devices(&rd);
-            return SoundIoErrorOpeningDevice;
+            rd.device_shared->probe_error = SoundIoErrorOpeningDevice;
```

Comment:
In this case, since we have a `SoundIoError` value in `err` from `from_lpwstr`, should we set the `probe_error` fields equal to that instead of `SoundIoErrorOpeningDevice`? A few more instances of this below as well.